tests/compatibility-test.py

```diff
@@ -264,75 +269,49 @@ def test_detached_actor(self):
     def setUpClass(cls):
-        if not utils.ray_service_supported(ray_version):
-            raise unittest.SkipTest("ray service is not supported")
-        # Ray Service is running inside a local Kind environment.
-        # We use the Ray nightly version now.
-        # We wait for the serve service ready.
-        # The test will check the successful response from serve service.
-        utils.delete_cluster()
-        utils.create_cluster()
-        images = [ray_image, kuberay_operator_image, kuberay_apiserver_image]
-        utils.download_images(images)
-        utils.apply_kuberay_resources(images, kuberay_operator_image, kuberay_apiserver_image)
-        utils.create_kuberay_service(
-            RayServiceTestCase.service_template_file, ray_version, ray_image)
+        if not utils.is_feature_supported(ray_version, CONST.RAY_SERVICE):
+            raise unittest.SkipTest(f"{CONST.RAY_SERVICE} is not supported")
+        K8S_CLUSTER_MANAGER.delete_kind_cluster()
+        K8S_CLUSTER_MANAGER.create_kind_cluster()
+        image_dict = {
+            CONST.RAY_IMAGE_KEY: ray_image,
+            CONST.OPERATOR_IMAGE_KEY: kuberay_operator_image
+        }
+        operator_manager = OperatorManager(image_dict)
+        operator_manager.prepare_operator()
 
     def test_ray_serve_work(self):
-        time.sleep(5)
-        curl_cmd = 'curl  -X POST -H \'Content-Type: application/json\' localhost:8000 -d \'["MANGO", 2]\''
-        utils.wait_for_condition(
-            lambda: utils.shell_run(curl_cmd) == 0,
-            timeout=15,
-        )
-        utils.create_kuberay_service(
-            RayServiceTestCase.service_serve_update_template_file,
-            ray_version, ray_image)
-        curl_cmd = 'curl  -X POST -H \'Content-Type: application/json\' localhost:8000 -d \'["MANGO", 2]\''
-        time.sleep(5)
-        utils.wait_for_condition(
-            lambda: utils.shell_run(curl_cmd) == 0,
-            timeout=60,
-        )
-        utils.create_kuberay_service(
-            RayServiceTestCase.service_cluster_update_template_file,
-            ray_version, ray_image)
-        time.sleep(5)
-        curl_cmd = 'curl  -X POST -H \'Content-Type: application/json\' localhost:8000 -d \'["MANGO", 2]\''
-        utils.wait_for_condition(
-            lambda: utils.shell_run(curl_cmd) == 0,
-            timeout=180,
-        )
-
+        """Create a RayService, send a request to RayService via `curl`, and compare the result."""
+        cr_event = utils.create_ray_service(
+            RayServiceTestCase.service_template_file, ray_version, ray_image)
+        # When Pods are READY and RUNNING, RayService still needs tens of seconds to be ready
+        # for serving requests. This `sleep` function is a workaround, and should be removed
+        # when https://github.com/ray-project/kuberay/pull/730 is merged.
+        time.sleep(60)
+        cr_event.rulesets = [RuleSet([CurlServiceRule()])]
+        cr_event.check_rule_sets()
```

Comment:
Do you think we need to add a test for updating RayService as well?

Reply:
Maybe we should have a test related to updating RayService in the future. Currently, I want to make sure the simplest test is stable enough. The original test logic is too weird. I will discuss with @shrekris-anyscale and @architkulkarni to design a thorough test plan for both RayService and RayJob.